### ISSUE TITLE
False-positive "Property does not exist" error in an `Object.prototype.hasOwnProperty` check block

### ISSUE THREAD

**Issue by Malix-off** [microsoft/TypeScript#56549]
### 🔎 Search Terms

Object Property, hasOwnProperty, if, check, block, scope, error, false-positive

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about the above tags

### ⏯ Playground Link

https://www.typescriptlang.org/play?#code/MYewdgzgLgBCBGArApsWBeGBvAUDfMADgE4iHLFQCeAjAFwwBEyAHgJbRtgDmjOAvjhxsAZjAAUeAgHoAVLKkFZMAErIoAV2JgGAMQCGAGwjJF+WdLMIUaAHQALfRADyAdzAAFUuUpVxAchIyCmoAJn8AShwI7DM5BQJzGABRFnI0ZAATBgA5EBTiUmJbMxhlNU1tBkZkwpBiBi9g3xhA7xCqcJhMkGQIGDAQWFYOWHAYanJWrCJ233oYaGIubgBuGH5-WxgAETZMmCoQDRgAW2R9MFagn2oafwB+cVCAVheaCL5EsstE0EgQIZkLZDCBuOJrKgoLYbh1QhFVgIcEA

### 💻 Code

```ts
const object = {
    property1: "existing"
}

if (
    /**
     * Return: False
     */
    object.hasOwnProperty('property2')
) {
    /**
     * Expected: No Error.
     * Return: "Error: Property 'property2' does not exist on type '{ property1: string; }'. Did you mean 'property1'?(2551)"
     */
    console.log(object.property2);
}

```

### 🙁 Actual behavior

> Error: Property 'property2' does not exist on type '{ property1: string; }'. Did you mean 'property1'?(2551)

### 🙂 Expected behavior

> No error

**Comment by MartinJohns** [comment]
Duplicate of #41915.
